### PR TITLE
[MAINTENANCE] Replace `Optional[Type]` with `Union[Type, None]` if no default exists (1 of 2)

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-1367
- See [this explanation](https://fastapi.tiangolo.com/python-types/#using-union-or-optional) for why we are only using `Optional` in method signatures for parameters with defaults
- Example for the only situation where `Optional` will remain:
```
def foo(bar: Optional[type] = None):
```
- Breaking up into 2 PRs due to number of changes required

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def __init__(
         bw_method: Optional[Union[str, float, Callable]] = None,
         include_estimator_samples_histogram_in_details: Union[str, bool] = False,
         truncate_values: Optional[
-            Union[str, Dict[str, Union[Optional[int], Optional[float]]]]
+            Union[str, Dict[str, Union[Union[int, None], Union[float, None]]]]
```
Comment (Kilo59) [inline great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py]:
```suggestion
            Union[str, Dict[str, Union[int, float, None]]]
```

File: great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py [hunk 2]
```diff
@@ -289,7 +289,7 @@ def include_estimator_samples_histogram_in_details(self) -> Union[str, bool]:
     @property
     def truncate_values(
         self,
-    ) -> Optional[Union[str, Dict[str, Union[Optional[int], Optional[float]]]]]:
+    ) -> Optional[Union[str, Dict[str, Union[Union[int, None], Union[float, None]]]]]:
```
Comment (Kilo59) [inline great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py]:
```suggestion
    ) -> Union[str, Dict[str, Union[int, float, None]], None]:
```

File: great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py [hunk 3]
```diff
@@ -718,7 +718,7 @@ def _get_truncate_values_using_heuristics(
         *,
         variables: Optional[ParameterContainer] = None,
         parameters: Optional[Dict[str, ParameterContainer]] = None,
-    ) -> Dict[str, Union[Optional[int], Optional[float]]]:
+    ) -> Dict[str, Union[Union[int, None], Union[float, None]]]:
```
Comment (Kilo59) [inline great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py]:
```suggestion
    ) -> Dict[str, Union[int, float, None]]:
```

**Review by alexsherstinsky — COMMENTED** [review]
While I am not seeing any errors, I feel that the verbose nature of the result outweighs the benefits.  To me, the succinct use of `Optional` to handle both cases, which maybe judged as an abuse of notation, is still acceptable as the interpretation is simple.  Hence, I am on the fence, and for that reason I would like to defer the decision to people who have more compelling arguments.  Thank you.

**Review by Kilo59 — APPROVED** [review]

**Pull request closed without merging**